pkg/server/registry/apps/validator.go

```diff
@@ -319,6 +325,19 @@ func checkMemory(memory v1.Memory, workloads map[string]v1.Container, specMemDef
 	return validationErrors
 }
 
+func validateMemoryRunFlags(memory v1.Memory, workloads map[string]v1.Container) *field.Error {
+	for key, value := range memory {
+		if key == "" {
+			continue
+		}
+		if _, ok := workloads[key]; !ok {
+			path := field.NewPath("spec", "memory")
+			return field.Invalid(path, fmt.Sprintf("%s=%v", key, *value), v1.ErrInvalidWorkload.Error())
+		}
```

Comment:
nit: Simplifies the code a bit and uses the path pattern used in `checkMemory`
```suggestion
		if _, ok := workloads[key]; !ok {
			path := field.NewPath("spec", "memory", key)
			return field.Invalid(path, *value, v1.ErrInvalidWorkload.Error())
		}
```

Reply:
any preference/thoughts?

Reply:
Should we mention in the error message that the container name doesn't exist/get defined the app? I think with the current error message, user still won't have an idea what's wrong the the specification.

Reply:
Change `*value` to `key`, as: `return field.Invalid(path, key, v1.ErrInvalidWorkload.Error())`

Reply:
Yeah that's right. We want this to be as clear as possible that the user defined a workload to change that does not exist.

Reply:
Additionally, I like the idea of collecting the errors here into a slice and returning them all. Similar to how `checkMemory` does.

Reply:
the error msg is now @thedadams @StrongMonkey 
```
 ERROR:  App.api.acorn.io "winter-lake" is invalid: spec.memory: Invalid value: "ng": workload name set by user does not exist
```

Reply:
> Additionally, I like the idea of collecting the errors here into a slice and returning them all. Similar to how `checkMemory` does.

It does to propagated up through `checkMemory` error  slice

Reply:
> Additionally, I like the idea of collecting the errors here into a slice and returning them all. Similar to how `checkMemory` does.

should be handled now for multiple errs